cli-commands/forgae-contracts/forgae-contracts.js

```diff
@@ -0,0 +1,130 @@
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const fs = require('fs-extra');
+const path = require('path');
+const git = require('simple-git/promise')();
+const Hjson = require('hjson');
+const opn = require('opn');
+const {
+  spawn
+} = require('promisify-child-process');
+
+const NODE_MODULES_LITERAL = 'node_modules';
+const CONTRACTS_AEPP_LITERAL = 'contracts-aepp';
+const YARN_LITERAL = 'yarn';
+const CONTRACTS_AEPP_GITHUB_URL = 'git+https://github.com/aeternity/aepp-contracts#develop';
+const CONTRACTS_SETTINGS_PATH = '/src/settings.js';
+const FORGAE_SETTINGS_PATH = '/cli-commands/forgae-contracts/settings.js';
+const DEFAULT_CONTRACTS_AEPP_URL = 'http://localhost:8080/';
+const DEFAULT_LOCAL_NODE_URL = 'http://localhost:3001/';
+const EXPORT_FILE_LITERAL = 'export default';
+const CONTRACTS_STARTING_SUCCESSFULLY = 'Compiled successfully';
+
+let nodeModulesPath;
+let contractAeppProjectPath;
+
+const run = async (options) => {
+  await mainFlow(options);
+};
+
+const mainFlow = async (options) => {
+  exec('npm list -g --depth 0', async function (error, stdout) {
+    const contractsAeppInstalled = checkGlobalModuleExistence(stdout, CONTRACTS_AEPP_LITERAL);
+
+    if (!contractsAeppInstalled || options.update) {
+      await installRepo();
+    }
+
+    await getFreshlyInstalledContractsAeppPath();
+
+    const yarnIsInstalled = checkGlobalModuleExistence(stdout, YARN_LITERAL);
+
+    if (yarnIsInstalled) {
+      await serveContractsAepp(options);
+    } else {
+      await installYarn();
+      await serveContractsAepp(options);
+    }
+  });
+};
+
+const checkGlobalModuleExistence = (stdout, moduleName) => {
+  const devDependenciesPath = stdout.split('\n')[0];
+  nodeModulesPath = path.join(devDependenciesPath, NODE_MODULES_LITERAL);
+  const modulesContent = fs.readdirSync(nodeModulesPath);
+  return modulesContent.includes(moduleName);
+};
+
+const installRepo = async () => {
+  console.log('====== Installing Contracts web Aepp ======');
+  await exec(`npm install -g ${CONTRACTS_AEPP_GITHUB_URL}`);
+};
+
+const installYarn = async () => {
+  console.log('====== Installing yarn ======');
```

Comment:
I think it would be better to install yarn when initing the project?

Reply:
I can't get your point. The yarn is installed if it is not installed at all, just before to serve the contracts aepp. As you can see here: 
```
    if (yarnIsInstalled) {
      await serveContractsAepp(options);
    } else {
      await installYarn();
      await serveContractsAepp(options);
    }
```
There is no initialization process as well.